src/steps/email-field-validation.ts

```diff
@@ -0,0 +1,106 @@
+/*tslint:disable:no-else-after-return*/
+
+import { BaseStep, Field, StepInterface } from '../core/base-step';
+import { Step, FieldDefinition, StepDefinition } from '../proto/cog_pb';
+import { Email, Inbox } from '../models';
+
+export class EmailFieldValidation extends BaseStep implements StepInterface {
+  private operators: string[] = ['Is Exactly', 'Contains', 'Does not Contain'];
+
+  protected stepName: string = 'Check a field on a Mailgun Email';
+  // tslint:disable-next-line:max-line-length
+  protected stepExpression: string = 'the (?<field>[a-zA-Z0-9_-]+) field of the email in position (?<position>[a-zA-Z0-9_-]+) from the inbox of (?<email>.+), (?<operator>.+) "(?<expectation>.+)"';
```

Comment:
Couple of notes on the expression:
- [ ] I recommend taking a look at the [Cog style guide, here](https://crank.automatoninc.com/#step-expressions).  Notable for this step: we should use a `should` keyword to make it clear the step is a validation.  It could also be tweaked to read a little more fluidly.
- [ ] I don't believe values (e.g. for the `expectation field` that themselves contain quotes will play well with regex capture groups that are surrounded by quotes.  This is why I've so far avoided using quotes around regex capture groups.
- [ ] One thing not in the style guide, but which we may want to add, is the recommendation that boundaries/validations for fields be encoded the regex itself.  For example...
  - You could avoid having to check any `field` values in code by specifying valid fields in the regex: `(?<field>(subject|body|from|etc))`
  - Same with the `operator` field: `(?<operator>(should contain|should not contain|should be))`
  - Also with the `position` field, we'd want to specify `\d+`

My recommendation, based on the fields you have here:
```regex
the (?<field>(subject|body|from)) of the (?<position>\d+)(?:(st|nd|rd|th))? mailgun email for (?<email>.+) (?<operator>(should contain|should not contain|should be)) (?<expectation>.+)
```

Reply:
Never thought to use the expression definition as part of the validation for enumerable fields i.e. operator, field. Thanks for helping out.

The recommended expression is really on point however I experienced a problem where if RegEx sits next to each other, the value you are attempting to pass to that pattern will be taken as one. 

Sample expression: the (regex1) (regex2)
And a yml step: the email should be
Then: "email should be" is parsed by (regex1) only while "should be" is for regex2.

Reply:
@amcpanaligan I tried pulling down your latest code and ran into the "step not found issue" that you may be referring to.  If I tweak the position regex from `(?<position>\d+)` to `(?<position>\\d+)` (with a double-escape), everything appears to work as expected!

Reply:
Thanks but what does the double `\\` or escape do/mean? And why did it fix the problem? How about for patterns without `\` like `(?<field>(subject|body|from))`. Can this be escaped as well?